### PR TITLE
Refactor: diff with router

### PR DESCRIPTION
**Pull request by dleard**
- [x] Move state variables into query string
- [x] Only fetch the xml for the reports being diffed

### CODE REVIEW

File: ggircs-app/app/pages/xml-diff.tsx [hunk 1]
```diff
@@ -1,102 +1,60 @@
...
-                  xmlFileName
-                  xmlFile
-                  ecccZipFileByZipFileId {
-                    zipFileName
-                  }
-                }
-              }
             }
           }
         }
```
Comment (pbastia) [inline ggircs-app/app/pages/xml-diff.tsx:35]: Maybe as a tech debt, we could avoid pulling the full list of reports and only verify if a report exists when the user types it in

File: ggircs-app/app/components/XmlDiff/RenderDiff.tsx [hunk 2]
```diff
@@ -1,35 +1,64 @@
...
+      setDiffText(
+        formatLines(diffLines(newText, oldText), {
+          // context is the max number of lines to show around a change. When collapsed, we set it to 1. Otherwise 10000 to ensure we get the whole document.
+          context: router.query.collapsed ? 1 : 10000,
+        })
+      );
+    else
+      setDiffText(
+        formatLines(diffLines(oldText, newText), {
+          context: router.query.collapsed ? 1 : 10000,
```
Comment (pbastia) [inline ggircs-app/app/components/XmlDiff/RenderDiff.tsx:55]: Maybe we could make this values configurable, for example in the props? Or declare it once on top of the file at least

**Review by pbastia — APPROVED** [review]
👍 👍 
Just a couple of discussion points, but we don't have to address them now

File: ggircs-app/app/pages/xml-diff.tsx (hunk 1 above)
Comment (dleard) [reply ggircs-app/app/pages/xml-diff.tsx:35]: captured in issue #344

**Review by pbastia — APPROVED** [review]

**Pull request merged**